### PR TITLE
TST: Remove some xfails in groupby tests

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/test_api_consistency.py [hunk 1]
```diff
@@ -82,13 +77,10 @@ def test_frame_consistency(request, groupby_func):
...
-        msg = "first and last are entirely different between Series and groupby"
-        request.node.add_marker(pytest.mark.xfail(reason=msg))
-    if groupby_func in ("cumcount", "corrwith"):
-        msg = "Series has no such method"
-        request.node.add_marker(pytest.mark.xfail(reason=msg))
+        # first and last are entirely different between Series and groupby
+        return
 
-    if groupby_func == "ngroup":
+    if groupby_func in ("cumcount", "corrwith", "ngroup"):
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_api_consistency.py:83]: ive had these corrwith xfails on my radar for a bit.  if we remove the xfail, should we also say "no" to #32293?

Comment (rhshadrach) [reply pandas/tests/groupby/test_api_consistency.py:83]: I don't think so; replacing them with the attribute check means that this test will still start failing if they are ever added, but without the perf impact of using an xfail.

**Review by phofl — APPROVED** [review]

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @rhshadrach

File: pandas/tests/groupby/test_api_consistency.py [hunk 2]
```diff
@@ -16,16 +14,13 @@
...
-def test_frame_consistency(request, groupby_func):
+def test_frame_consistency(groupby_func):
     # GH#48028
     if groupby_func in ("first", "last"):
-        msg = "first and last are entirely different between frame and groupby"
-        request.node.add_marker(pytest.mark.xfail(reason=msg))
-    if groupby_func in ("cumcount",):
-        msg = "DataFrame has no such method"
-        request.node.add_marker(pytest.mark.xfail(reason=msg))
+        # first and last are entirely different between frame and groupby
```
Comment (mroeschke) [inline pandas/tests/groupby/test_api_consistency.py:20]: For these cases where nothing is tested, I would prefer using `pytest.skip(reason)` just so it's clear that nothing is supposed to be tested

Comment (rhshadrach) [reply pandas/tests/groupby/test_api_consistency.py:20]: Makes sense - I'll do a follow up.

Comment (mroeschke) [reply pandas/tests/groupby/test_api_consistency.py:20]: Thanks (totally minor)

**Pull request merged**